Source/Core/DolphinQt/Config/Mapping/GCPadWiiUConfigDialog.cpp

```diff
@@ -24,28 +24,30 @@ GCPadWiiUConfigDialog::GCPadWiiUConfigDialog(int port, QWidget* parent)
   ConnectWidgets();
 }
 
+GCPadWiiUConfigDialog::~GCPadWiiUConfigDialog()
+{
+  GCAdapter::SetAdapterCallback(nullptr);
+}
+
 void GCPadWiiUConfigDialog::CreateLayout()
 {
   setWindowTitle(tr("GameCube Adapter for Wii U at Port %1").arg(m_port + 1));
 
-  const bool detected = GCAdapter::IsDetected();
   m_layout = new QVBoxLayout();
-  m_status_label = new QLabel(detected ? tr("Adapter Detected") : tr("No Adapter Detected"));
+  m_status_label = new QLabel();
   m_rumble = new QCheckBox(tr("Enable Rumble"));
   m_simulate_bongos = new QCheckBox(tr("Simulate DK Bongos"));
   m_button_box = new QDialogButtonBox(QDialogButtonBox::Ok);
 
+  UpdateAdapterStatus();
+
+  GCAdapter::SetAdapterCallback([this] { QMetaObject::invokeMethod(this, "UpdateAdapterStatus"); });
```

Comment:
Isn't there any other method to call `UpdateAdapterStatus` that doesn't rely on a magic string?

Reply:
I wasn’t aware of it, but Qt 5 apparently allows using member functions pointers for the signal/slots system instead of strings (https://wiki.qt.io/New_Signal_Slot_Syntax).
So `QMetaObject::invokeMethod(this, &GCPadWiiUConfigDialog::UpdateAdapterStatus);` actually works (and is apparently fully type-safe). I will change the code to use it.